src/lightning/fabric/strategies/xla_fsdp.py

```diff
@@ -0,0 +1,392 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from contextlib import contextmanager
+from pathlib import Path
+from typing import Any, Callable, Dict, Generator, List, Literal, Optional, Tuple, Union
+
+import torch
+from torch import Tensor
+from torch.nn import Module
+from torch.optim import Optimizer
+
+from lightning.fabric.accelerators import Accelerator
+from lightning.fabric.accelerators.xla import _XLA_GREATER_EQUAL_2_1
+from lightning.fabric.plugins.io.checkpoint_io import CheckpointIO
+from lightning.fabric.plugins.precision import Precision
+from lightning.fabric.strategies import _StrategyRegistry, XLAStrategy
+from lightning.fabric.strategies.fsdp import _apply_filter
+from lightning.fabric.strategies.strategy import _BackwardSyncControl, _validate_keys_for_strict_loading
+from lightning.fabric.utilities.imports import _TORCH_GREATER_EQUAL_2_0
+from lightning.fabric.utilities.rank_zero import rank_zero_warn
+from lightning.fabric.utilities.types import _PATH, Optimizable
+
+
+class XLAFSDPStrategy(XLAStrategy):
+    """Strategy for training multiple XLA devices using the
+    :func:`torch_xla.distributed.xla_fully_sharded_data_parallel.XlaFullyShardedDataParallel` method.
+
+    .. warning::  This is an :ref:`experimental <versioning:Experimental API>` feature.
+
+    For more information check out https://github.com/pytorch/xla/blob/master/docs/fsdp.md
+    """
+
+    def __init__(
+        self,
+        accelerator: Optional[Accelerator] = None,
+        parallel_devices: Optional[List[torch.device]] = None,
+        checkpoint_io: Optional[CheckpointIO] = None,
+        precision: Optional[Precision] = None,
+        state_dict_type: Literal["full", "sharded"] = "sharded",
+        sync_module_states: bool = True,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(
+            accelerator=accelerator,
+            parallel_devices=parallel_devices,
+            checkpoint_io=checkpoint_io,
+            precision=precision,
+            sync_module_states=sync_module_states,
+        )
+        self._backward_sync_control = _XLAFSDPBackwardSyncControl()
+
+        self._fsdp_kwargs = kwargs
+        self._state_dict_type = state_dict_type
+
+    def setup_module_and_optimizers(
+        self, module: Module, optimizers: List[Optimizer]
+    ) -> Tuple[Module, List[Optimizer]]:
+        """Returns NotImplementedError since for XLAFSDP optimizer setup must happen after module setup."""
+        raise NotImplementedError(
```

Comment:
The XLA fsdp strategy does not support `use_orig_params` like the PyTorch native one does, correct?